LSP Rewrite Step 1: fix many bugs, more LSP test coverage! (#3521)
graphql/graphiql: packages/vscode-graphql/README.md

For more help with configuring the language server,
[the language server readme](https://github.com/graphql/graphiql/tree/main/packages/graphql-language-service-server/README.md)
is the source of truth for all settings used by all editors which use the
language server.

This includes LSP settings provided by extensions like `vscode-graphql`, nvim, etc.

> There are a number of configurations that can be provided from both editor settings or the graphql config file, and the editor setting takes precedence, to allow users to override their graphql config file settings in a user context as needed.

Reviewer: ```suggestion
There are a number of configurations that can be provided from editor settings and/or the graphql config file, with the editor settings taking precedence, so users can override their graphql config file settings in a user context, as needed.
```
